### PR TITLE
Document default and third party providers in YAML reference

### PR DESCRIPTION
**Pull request by AaronFriel**
### Proposed changes

Documents how to set a default provider for multiple resources and how to use `pluginDownloadURL` with third party providers, whether community or customer managed.

### CODE REVIEW

File: content/docs/languages-sdks/yaml/yaml-language-reference.md [hunk 1]
```diff
@@ -116,23 +116,34 @@ There are at least two reasons to explicitly define providers in YAML, or explic
 1. Using explicit versions enables pinning the dependencies used, a technique used to improve build reliability.
 2. Using explicit providers enables controlling the options for providers used by each resource, as described in [Unlock Programmatic Control by Disabling Default Providers](/blog/disable-default-providers/).
 
-#### Resource version
+If a specific provider version is necessary, we prefer the approaches in the order described here.
```
Comment (AaronFriel) [inline content/docs/languages-sdks/yaml/yaml-language-reference.md:119]: Want users to fall into "pit of success" here, a program with default providers is more likely to be correct than one with explicit providers passed to multiple resources or setting these options on individual resources.

File: content/docs/languages-sdks/yaml/yaml-language-reference.md [hunk 2]
```diff
@@ -116,23 +116,34 @@ There are at least two reasons to explicitly define providers in YAML, or explic
...
 ```
 
+In this example the `my-bucket` resource, and any other AWS resources, uses the default AWS provider version 6.45.0.
+
+Only one provider of a given `type` can be declared as the default provider.
+
 #### Explicit provider
 
-To create an explicit provider instance, preferably with a specific version, use the [`resources`](#resources) section. For the `type` property, prefix the name of the provider with `pulumi:providers`.
+To declare an explicit provider instance for specific resources, declare a provider resource the [`resources`](#resources) section. For the `type` property, prefix the name of the provider with `pulumi:providers`.
```
Comment (AaronFriel) [inline content/docs/languages-sdks/yaml/yaml-language-reference.md:146]: Bit of a nit fixed here, Pulumi programs don't create resources, they declare them.

Comment (AaronFriel) [inline content/docs/languages-sdks/yaml/yaml-language-reference.md:146]:
```suggestion
To declare an explicit provider instance for specific resources, declare a provider resource in the [`resources`](#resources) section. For the `type` property, prefix the name of the provider with `pulumi:providers`.
```

File: content/docs/languages-sdks/yaml/yaml-language-reference.md [hunk 3]
```diff
@@ -116,23 +116,34 @@ There are at least two reasons to explicitly define providers in YAML, or explic
 1. Using explicit versions enables pinning the dependencies used, a technique used to improve build reliability.
 2. Using explicit providers enables controlling the options for providers used by each resource, as described in [Unlock Programmatic Control by Disabling Default Providers](/blog/disable-default-providers/).
 
-#### Resource version
+If a specific provider version is necessary, we prefer the approaches in the order described here.
+
+#### Default provider
+
+Declaring a default provider simplifies configuring a provider version across all resources for that provider, as all subsequent resources will
```
Comment (justinvp) [inline content/docs/languages-sdks/yaml/yaml-language-reference.md:123]: ... will what?

Comment (AaronFriel) [reply content/docs/languages-sdks/yaml/yaml-language-reference.md:123]:
```suggestion
Declaring a default provider simplifies configuring a provider version across all resources for that provider.
```

File: content/docs/languages-sdks/yaml/yaml-language-reference.md [hunk 4]
```diff
@@ -158,6 +169,42 @@ resources:
...
+      ...
+    options:
+      version: 5.6.0
+```
+
+#### Third party providers
+
+Third party providers may require a `pluginDownloadURL` option for Pulumi to acquire the provider plugin. The publisher of that provider should provide this URL, following our guide for [Authoring & Publishing](/docs/using-pulumi/pulumi-packages/how-to-author/#publish-your-package).
+
+All of the above examples setting a `version` with a default provider, explicit provider, or on individual resources may be combined with the `pluginDownloadURL` option to use a third party provider.
```
Comment (AaronFriel) [inline content/docs/languages-sdks/yaml/yaml-language-reference.md:192]:
```suggestion
All of the above examples setting a `version` with a default provider, explicit provider, or on individual resources may be combined with the `pluginDownloadURL` option to use a third party provider.
```

**Review by justinvp — APPROVED** [review]

**Pull request merged**